runner/src/main/kotlin/com/cosmotech/runner/service/RunnerService.kt

```diff
@@ -256,6 +287,57 @@ class RunnerService(
               ?: mutableListOf())
     }
 
+    @Suppress("NestedBlockDepth")
+    private fun constructParametersValuesFromRunTemplateAndParent(
+        parentId: String?,
+        solution: Solution?,
+        runTemplate: RunTemplate?,
+        parent: Runner,
+        runner: Runner
+    ): MutableList<RunnerRunTemplateParameterValue> {
+      val parametersValuesList = mutableListOf<RunnerRunTemplateParameterValue>()
+      logger.debug("Copying parameters values from parent $parentId")
+
+      logger.debug("Getting runTemplate parameters ids")
+      val runTemplateParametersIds =
+          solution
+              ?.parameterGroups
+              ?.filter { parameterGroup ->
+                runTemplate?.parameterGroups?.contains(parameterGroup.id) == true
+              }
+              ?.flatMap { parameterGroup -> parameterGroup.parameters ?: mutableListOf() }
+      if (!runTemplateParametersIds.isNullOrEmpty()) {
+        val parentParameters = parent.parametersValues?.associate { it.parameterId to it }
+        val runnerParameters = runner.parametersValues?.associate { it.parameterId to it }
+        // TODO Handle default value
```

Comment:
Comment to be removed as discussed IRL

Reply:
As last minute conclusion, TODO was modified to contain "Handle default value" context.